examples/monitoring/main.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/capatazlib/go-capataz/s"
+)
+
+// default port for prometheus metrics server
+const metricsHTTPAddr = "0.0.0.0:8080"
+
+// stopOnSignal will wait for the program to receive a SIGTERM and stop the
+// given supervisor
+func stopOnSignal(sup s.Supervisor) {
+	done := make(chan os.Signal, 1)
+	signal.Notify(done, syscall.SIGTERM)
+	<-done
+	err := sup.Stop()
+	fmt.Println(err)
+}
+
+// This program runs a prometheus metrics server and a few goroutines that print
+// greetings in the standard output. It showcases how to build an app by
+// composing all the components in a tree shape. All the wiring happens before
+// we run any business logic.
+//
+// This API gives restart logic for any failures that happen in the goroutines
+// and also forces us to build our application as a group of sub-systems and
+// goroutine workers.
+//
+func main() {
+	// Setup the logging mechanisms
+	log, logEventNotifier := newLogEventNotifier()
+
+	// Build a supervision tree that runs the Prometheus HTTP server
+	prometheusSpec, promEventNotifier := newPrometheusSpec("prometheus", metricsHTTPAddr)
```

Comment:
This is 100% personal preference and you may ignore it, but I think that establishing a pattern is important. Doing so will make reading future code easier (i.e. require less cognitive load???).

As you did with the greeter and logger (maybe consider calling it `greeterEventNotifier()` instead??), would you consider a `newPrometheusEventNotifier()` followed by a `NewPrometheusSPec()` that takes the `prometheusEventNotfier()`?

There is nothing _wrong_ with what you have. It's just that because it is different, I had to go back and forth to see what was going on. 🤷‍♂

Reply:
I agree, I'm going to make the functions look alike as much as possible if they are doing the same thing